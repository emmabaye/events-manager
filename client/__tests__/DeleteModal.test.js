import ConnectedDeleteModal from '../components/DeleteModal.jsx';

describe('DeleteModal Component', () => {
  global.localStorage = {
    getItem: (str) => 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9' +
    '.eyJpZCI6NzUsInJvbGUiOiJhZG1pbiIsImlhdCI6MTUxMjEyMzIyMSwiZXhwIjoxNTEyMjA5NjIxfQ' +
    '.IwykmkkET5VmBvTEyPPXfMOq611ITQaoh-9wV11mJgE'
  };

  const initialState = {
    eventReducer: {
      userEvents: []
    },
    centerReducer: {
      allCenters: []
    }
  };

  const store = mockStore(initialState);

  const props = {
    itemId: "1",
    item: "center"
  };

  it('it should render DeleteModal component', () => {
    const wrapper = shallow(<Provider store={store}><ConnectedDeleteModal {...props} /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should handle onClick event', () => {
    const wrapper = mount(<Provider store={store}><ConnectedDeleteModal {...props} /></Provider>);
    let DeleteButton = wrapper.find('.btn-danger');
    DeleteButton.simulate('click');
  });
});
