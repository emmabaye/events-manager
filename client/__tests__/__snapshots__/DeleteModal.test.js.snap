// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component it should render DeleteModal component 1`] = `
<Connect(DeleteModal)
  item="center"
  itemId="1"
/>
`;
